scheduling/priority_scheduling.py

```diff
@@ -0,0 +1,147 @@
+"""
+A Priority Scheduling algorithm
+
+https://en.wikipedia.org/wiki/Scheduling_(computing)#Priority_scheduling
+"""
+
+from operator import attrgetter
+from typing import List
+
+
+class Process:
+    def __init__(self, name: str, arrival_time: int, burst_time: int, priority: int):
+        self.name = name
+        self.arrival_time = arrival_time
+        self.burst_time = burst_time
+        self.priority = priority
+        self.waiting_time: int = 0
+        self.turn_around_time: int = 0
+
+    def __repr__(self):
+        """
+        >>> Process("Name", 1, 2, 3)
+        Process(name='Name', arrival_time=1, burst_time=2, priority=3)
+        """
+        return (
+            f"{self.__class__.__qualname__}(name='{self.name}', "
+            f"arrival_time={self.arrival_time}, burst_time={self.burst_time}, "
+            f"priority={self.priority})"
+        )
+
+    def __str__(self):
+        r"""
+        >>> str(Process("Name", 1, 2, 3))
+        'Name\t\t0\t\t0'
+        """
+        return f"{self.name}\t\t{self.waiting_time}\t\t{self.turn_around_time}"
+
+
+def create_processes(number_of_processes: int = 0) -> List[Process]:
+    def get_data_for_process(process_number: int) -> Process:
+        return Process(
+            name=str(
+                input(f"Enter Name of process {process_number}: ").strip()
+                or process_number
+            ),
+            arrival_time=int(input("Enter Arrival time: ").strip() or process_number),
+            burst_time=int(input("Enter Burst time: ").strip() or process_number),
+            priority=int(input("Enter Priority: ").strip() or process_number),
+        )
+
+    number_of_processes = number_of_processes or int(
+        input("Enter number of processes: ").strip() or 3
+    )
+    return [get_data_for_process(i) for i in range(number_of_processes)]
+
+
+def priority_schedule(processes: List[Process]) -> List[Process]:
+    """
+    >>> [process.name for process in priority_schedule([
+    ...     Process("P1", 0, 11, 3),
+    ...     Process("P2", 5, 28, 1),
+    ...     Process("P3", 12, 2, 4),
+    ...     Process("P4", 2, 10, 2),
+    ...     Process("P5", 9, 16, 4),
+    ... ])]
+    ['P1', 'P2`, 'P4', 'P3', 'P5']
```

Comment:
This doctest should correspond to your table.

Reply:
But the outcome (https://travis-ci.com/github/TheAlgorithms/Python/builds/178562807#L804) fails to match our expectations.

You can run this test on your own machine with the command `python3 -m doctest -v priority_scheduling.py`